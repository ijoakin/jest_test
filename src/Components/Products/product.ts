export interface Product {
  Id: number;
  Description: string;
  Price: number;
}