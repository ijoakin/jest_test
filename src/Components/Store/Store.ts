import { createStore } from 'redux';
import { ProductReducer } from './Reducers/ProductReducer';

export const store = createStore(ProductReducer);