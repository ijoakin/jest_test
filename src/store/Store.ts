import { createStore } from 'redux';
import { initialState, ProductReducer } from './Reducers/ProductReducer';
import {composeWithDevTools} from 'redux-devtools-extension';

const composeEnhancers = composeWithDevTools({});

export const store = createStore(ProductReducer, initialState, composeEnhancers());