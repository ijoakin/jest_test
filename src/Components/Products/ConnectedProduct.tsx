import { connect } from "react-redux";
import { ListProducts } from ".";
import { ProductsState } from "../Store/Reducers/ProductReducer";
import { Product } from "./product";

export interface ListProductsProps {
  prod: Product;
}

export const mapStateToProps = (state: ProductsState): ListProductsProps => {
  return {
    prod: state.product,
  };
};

export default connect(mapStateToProps)(ListProducts);
