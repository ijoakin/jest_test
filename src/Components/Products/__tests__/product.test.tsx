import React from "react";
import rendered from "react-test-renderer";
import { Products } from "../index";
import {
  screen,
  fireEvent,
  render,
  cleanup,
  waitFor,
} from "@testing-library/react";
import { Product } from "../product";
import { Provider } from "react-redux";
import {
  initialState,
  ProductsState,
} from "../../../store/Reducers/ProductReducer";
import { store } from "../../../store/Store";
import { mapStateToProps } from "../ConnectedProduct";

describe("First test", () => {
  beforeAll(() => {
    console.log("before All");
  });

  beforeEach(() => {
    console.log("before Each");
  });

  afterAll(() => {
    console.log("after All");
  });

  afterEach(() => {
    console.log("after Each");
  });

  it("should Execute product component", () => {
    expect(2 + 2).toEqual(4);
  });

  it("should work", () => {
    expect(2 + 2).toEqual(4);
  });

  it("renders as expected", () => {
    // eslint-disable-next-line @typescript-eslint/no-unused-vars
    const prod: Product = {
      Description: "description",
      Id: 10,
      Price: 12,
    };

    const tree = rendered
      .create(
        <Provider store={store}>
          <Products prod={prod}></Products>
        </Provider>
      )
      .toJSON();

    console.log(tree);
    expect(tree).toMatchSnapshot();
  });

  describe("mapstateToPorps", () => {
    it("should map state to props correctly", () => {
      const prod: Product = {
        Id: 10,
        Description: "Description 10",
        Price: 13,
      };

      const appState: ProductsState = {
        product: prod,
        listProducts:[],
        total:0
      };
      const componentState = mapStateToProps(appState);

      expect(componentState.prod).toEqual(appState.product);
    });
  });
});
