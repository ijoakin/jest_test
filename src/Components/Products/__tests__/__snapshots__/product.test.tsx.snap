// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`First test renders as expected 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <form
        className=""
        noValidate={true}
      >
        <div
          className="form-group"
        >
          <label
            className="form-label"
            htmlFor="exampleForm.ControlInput1"
          >
            Id
          </label>
          <input
            className="form-control"
            id="exampleForm.ControlInput1"
            placeholder="1231"
            type="text"
            value={10}
          />
        </div>
        <div
          className="form-group"
        >
          <label
            className="form-label"
            htmlFor="exampleForm.ControlInput1"
          >
            new field
          </label>
          <input
            className="form-control"
            id="exampleForm.ControlInput1"
            placeholder="1231"
            type="text"
            value="test"
          />
        </div>
        <div
          className="form-group"
        >
          <label
            className="form-label"
            htmlFor="exampleForm.ControlInput1"
          >
            Description
          </label>
          <input
            className="form-control"
            id="exampleForm.ControlInput1"
            placeholder="1231"
            type="text"
            value="description"
          />
        </div>
        <div
          className="form-group"
        >
          <label
            className="form-label"
            htmlFor="exampleForm.ControlInput1"
          >
            Price
          </label>
          <input
            className="form-control"
            id="exampleForm.ControlInput1"
            placeholder="1231"
            type="text"
            value={12}
          />
        </div>
      </form>
      <button
        onClick={[Function]}
      >
        Get Product
      </button>
    </div>
  </div>
</div>
`;
